source/versioned-api/versioned-api.rst

```diff
@@ -0,0 +1,273 @@
+options. As with all other commands, drivers MUST inherit the API version from
+the client. Specifying API versioning options in the command document and
+declaring an API version on the client is not supported. Drivers MUST document
+that the behaviour of the command helper is undefined in this case.
+
+
+Design Rationale
+================
+
+No URI Options
+--------------
+
+Since changing the API version can cause the application to behave differently,
+drivers MUST NOT allow users to change the declared API version without
+deploying code changes. This ensures that users don't copy a connection string
+with a declared API version that may be different from what their application
+expects. A URI option can be added later if we realise our users need it, while
+the opposite is not easily accomplished.
+
+
+Don't Allow Overriding the Declared API Version
+-----------------------------------------------
+
+While users are used to overriding options like read preference, read concern,
+and write concern in ``MongoDatabase`` and ``MongoCollection`` objects, or on an
+operation level, we explicitly decided against this for the declared API
+version. With a single API version available to start, we can't anticipate what
+use cases users may have to override the API version. Not including this feature
+at the beginning allows us to gather feedback on use cases and add the features
+users are looking for. On the other hand, adding the ability to override the
+declared API version can't be undone until a future major release, which is
+almost impossible to accomplish across all drivers.
+
+
+Generic Command Helper Behaviour
+--------------------------------
+
+The runCommand helper is a way for the user to run a native command with the
+driver doing little to no inspection in the command. This allows users to run
+arbitrary commands that may not have helpers in the driver, or to pass options
+that are not supported by the driver version they are currently using. Commands
+run using this helper do not inherit any ``readConcern`` or ``writeConcern``
+options that may have been set on the ``MongoClient`` or ``MongoDatabase``
+objects.
+
+However, the declared API version is a different case. We are introducing this
+feature to give users a certain peace of mind when upgrading driver or server
+versions, by ensuring that their code will continue to show the same behaviour
+they've gotten used to. This includes all commands run using the generic command
+helper. Thus, the helper will inherit the API version declared on the client.
+
+
+Hardcode supported versions in drivers
+--------------------------------------
+
+Since a new API version might require driver changes (e.g. to account for
+removed commands), we don't yet know what changes drivers must make for a future
+version. Until we do, we must prevent users from choosing any unknown API
+version.
+
+
+Backward Compatibility
```

Comment:
Should we mention the related policy changes in the spec at all? I imagine in the future we'll add spec language for an "experimental" feature, so we'll need to have that term defined in the specs somewhere.

Reply:
Since the policy is not public, I wanted to avoid mentioning it here. I can add a note about it for all driver engineers that will be reading this.

Reply:
Yeah true. I think whenever we do add the first "experimental" API to the specifications we can just tackle this issue then.